### DISCUSSION TITLE
Does every API route include _app.tsx in it's first load JS?

### DISCUSSION THREAD

**Discussion by JoeRoddy** [vercel/next.js#50002] [Help]
### Summary

I have some global UI components and utilities that get imported into _app.tsx (think Navbar, initializing logger, etc.)

I wouldn't of expected this, but as new dependencies are imported into `_app.tsx`, the First Load JS of every one of our APIs grow proportionately.  I thought `_app.tsx` was only supposed to be relevant on the client side?

Any explanation would be greatly appreciated.

![image](https://github.com/vercel/next.js/assets/16690442/ed59c050-c3b5-4e7f-aea2-0c56a1f6568f)

**Comment by devjiwonchoi** [comment]
The `_app.tsx` is executed on both the server-side and the client-side. On the first load `_app.tsx` will contribute to the 
JS bundle for SSR, including APIs. After the initial render of the client receives HTML, the `_app.tsx` executes again for the CSR. 

As a result, `_app.tsx` bundle will be affected by both client-side and server-side dependencies.
To reduce the first load JS, you could use [lazy loading](https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading) or [dynamic import of components](https://nextjs.org/learn/seo/improve/dynamic-imports), 
furthermore checkout for advanced usage of the [tree shaking](https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking) method.

In summary, it is natural that the `_app.tsx` is affected by your APIs.

**Reply by JoeRoddy** [reply]
Thanks for the reply!  Maybe I was misunderstanding what `_app.tsx` is, I didn't think it would run when I call an API, is that what you're saying?

 So where, if anywhere, is the appropriate place to import global dependencies that are only relevant on the client side? (Think initialization of a client side logging library)

Ideally that wouldn't have any effect on the bundle size for a totally unrelated API endpoint.

**Reply by JoeRoddy** [reply]
Re-reading your answer, I think maybe I was unclear with my initial question.

> In summary, it is natural that the _app.tsx is affected by your APIs.

I was asking the opposite.  Is it normal for APIs to grow in size when `_app.tsx` grows?

**Reply by devjiwonchoi** [reply]
My bad, I misunderstood your question. The short answer is no, and as far as I know the _app.tsx should not affect your APIs bundle size, only the other way around. 

I assume that the dependencies from the client side were shared with API pages or maybe 
the changes in client side pages affected the project's overall structure which triggered to 
optimize API bundle differently in response to the change, and cause to increase in API bundle size. 

I'm sure that it is not the case that the _app.tsx is directly affecting the APIs' bundle size. 
For further inspection you could use [bundlephobia](https://bundlephobia.com/) and [next/bundle-analyzer](https://www.npmjs.com/package/@next/bundle-analyzer) which is published by official Vercel.

**Reply by JoeRoddy** [reply]
Ok, thanks!  I think it might be a bug?

https://github.com/vercel/next.js/issues/45607

I just tried it on a fresh project with create-next-app and confirmed it's happening there too.   

Did a build, _app.tsx and all apis had the same size.  

Imported lodash into `_app.tsx` and it and all APIs grew by the same amount.

**Reply by devjiwonchoi** [reply]
Interesting.. I'll keep an eye on it until it is resolved. Thank you.
